Civi/Civioffice/DocumentEditorManager.php

```diff
@@ -122,67 +109,86 @@ public function saveNewEditor(
-    $id = (array_pop($editorIds) ?? 0) + 1;
-    $this->persistEditor($id, $name, $active, $fileExtensions, $type::getName(), $typeConfig);
-  }
-
-  /**
-   * @return list<int>
-   */
-  private function getEditorIds(): array {
-    return array_keys($this->getEditorNames());
+    $this->persistEditor(NULL, $name, $active, $fileExtensions, $type::getName(), $typeConfig);
   }
 
   /**
-   * @return array<int, string>
-   *   Mapping of editor ID to editor name.
+   * @phpstan-param whereT $where
+   *
+   * @return list<DocumentEditor>
+   *   Editors without configured file extensions are at the end.
+   *
+   * @throws \CRM_Core_Exception
    */
-  private function getEditorNames(): array {
-    if (NULL === $this->editorNames) {
-      /** @var array<int, string>|null $editorNames */
-      $editorNames = \Civi::settings()->get('civioffice_editors');
+  private function getEditors(array $where = []): array {
+    $editors = [];
+    $editorsWithoutExtensions = [];
 
-      $this->editorNames = is_array($editorNames) ? $editorNames : [];
+    $configurations = CiviofficeDocumentEditor::get(FALSE)->setWhere($where)->execute();
+    /** @phpstan-var array{
+     *   id: int,
+     *   name: string,
+     *   is_active: bool,
+     *   file_extensions: list<string>,
+     *   type: string,
+     *   type_config: array<string, mixed>,
+     * } $configuration */
+    foreach ($configurations as $configuration) {
+      $type = $this->typeContainer->get($configuration['type']);
+      $editor = new DocumentEditor($configuration, $type);
+      if ([] === $editor->getFileExtensions()) {
+        $editorsWithoutExtensions[] = $editor;
+      }
+      else {
+        $editors[] = $editor;
+      }
     }
 
-    return $this->editorNames;
+    return [...$editors, ...$editorsWithoutExtensions];
   }
 
   /**
    * @param list<string> $fileExtensions
    * @param array<string, mixed> $typeConfig
+   *
+   * @throws \CRM_Core_Exception
    */
   private function persistEditor(
-    int $id,
+    ?int $id,
     string $name,
     bool $active,
     array $fileExtensions,
     string $typeName,
     array $typeConfig
   ): void {
     $configuration = [
-      'active' => $active,
-      'fileExtensions' => $fileExtensions,
+      'name' => $name,
+      'is_active' => $active,
+      'file_extensions' => $fileExtensions,
       'type' => $typeName,
-      'typeConfig' => $typeConfig,
+      'type_config' => $typeConfig,
     ];
 
-    \Civi::settings()->set('civioffice_editor_' . $id, $configuration);
-    $this->editorNames[$id] = $name;
-    \Civi::settings()->set('civioffice_editors', $this->editorNames);
+    if (NULL === $id) {
+      $id = CiviofficeDocumentEditor::create(FALSE)
+        ->setValues($configuration)
+        ->execute()
+        ->single()['id'];
+    }
+    else {
+      CiviofficeDocumentEditor::update(FALSE)
+        ->addWhere('id', '=', $id)
+        ->setValues($configuration)
+        ->execute();
+    }
```

Comment:
This can be replaced with a single
```php
CiviofficeDocumentEditor::save()
          ->addRecord($configuration)
          ->setMatch(['id'])
          ->execute();
```